src/cljs/kixi/hecuba/tabs/hierarchy/datasets.cljs

```diff
@@ -328,17 +326,18 @@
   (reify
     om/IInitState
     (init-state [_]
-      {:dropdown-chan (chan)})
+      {:dropdown-chan (chan)
+       :name (:name selected-dataset)})
     om/IWillMount
     (will-mount [_]
       (go-loop []
         (let [{:keys [dropdown-chan]} (om/get-state owner)
-              {:keys [path value]}    (<! dropdown-chan)]
-          (om/update! selected-dataset path (if (= path [:field])
-                                              (str value "~" (-> (filter #(= (:value %) value) available-fields)
-                                                                 first
-                                                                 :unit))
-                                              value)))
+              {:keys [path value]}    (<! dropdown-chan)
+              v (if (= path [:field])
+                  (str value "~" (-> (filter #(= (:value %) value) available-fields) first :unit))
```

Comment:
not a problem in this instance now, but if we were to add a function after :unit I'd be thrown off by doing a seq thread first (->) rather than thread last (->>). So I'd prefer it to look like this:

```
(->> available-fields (filter #(= (:value %) value)) first :unit)
```